packages/core-react/src/components/AutoConnect.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Chain } from 'wagmi';
+import type { WalletAdapterData } from '@reef-knot/types';
+import { useAutoConnect } from '../hooks/useAutoConnect';
+
+export const AutoConnect = ({
```

Comment:
Why do needs an entire component if it only uses 1 hook? Component parameters correspond to hook parameters. Wouldn't it be easier to use one hook?

Reply:
Seems, it is because hook is using providers that are right above `AutoConnect` component and contexts cannot be accessed in component above. But maybe this is an indicator that autoconnect logic should be moved inside some of contexts.

Reply:
`useAutoConnect` hook uses `AcceptTermsModalContext` to control the additional modal, which asks a user to accept Terms of service before auto-connection. Therefore, `useAutoConnect` must be called from a place, wrapped into `AcceptTermsModalContext`. I decided to move it to a separate component.